drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -20,8 +21,68 @@
 
 LOG_MODULE_REGISTER(siwx91x_wifi);
 
+#define SIWX91X_INTERFACE_MASK (0x03)
+
 NET_BUF_POOL_FIXED_DEFINE(siwx91x_tx_pool, 1, _NET_ETH_MAX_FRAME_SIZE, 0, NULL);
 
+static inline int siwx91x_convert_mode(sl_wifi_interface_t sl_iface)
+{
+	switch (sl_iface) {
+	case SL_WIFI_CLIENT_2_4GHZ_INTERFACE:
+		return WIFI_STA_MODE;
+	case SL_WIFI_AP_2_4GHZ_INTERFACE:
+		return WIFI_AP_MODE;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int siwx91x_nwp_reboot(uint8_t req_mode)
+{
+	sl_wifi_device_configuration_t nwp_config;
+	sl_wifi_interface_t interface;
+	sl_status_t status;
+
+	interface = sl_wifi_get_default_interface();
+	switch (req_mode) {
+	case WIFI_STA_MODE:
+		if (interface == (SL_WIFI_CLIENT_INTERFACE | SL_WIFI_2_4GHZ_INTERFACE)) {
```

Comment:
What 2.4GHz is doing i the picture? You probably want to say: `FIELD_GET(SIWX91X_INTERFACE_MASK, interface) == SL_WIFI_CLIENT_INTERFACE`.